packages/minim-api-description/lib/define-value-of.js

```diff
@@ -88,33 +90,45 @@ module.exports = (namespace) => {
     return undefined;
   }
 
-  // Create instance of leaf element (Boolean, String, Number, Null)
-  function valueOfLeaf(e, options) {
+  // TODO e instanceof EnumElement fails because of dependency injection
+  // checking for e.element === 'enum' as a temporary  walkaround
+  const isEnumElement = e => (e.element === 'enum' || e instanceof EnumElement);
+
+  function valueOfAny(e, options, recurse) {
     const opts = updateTypeAttributes(e, options);
     if (e.content) {
-      return wrapResult(e.content, 'content', opts);
+      return wrapResult(recurse(e.content, clearFlag(SOURCE_FLAG, opts)), 'content', opts);
     }
     const sample = findFirstSample(e);
     if (sample) {
-      return wrapResult(sample.content, 'sample', opts);
+      return wrapResult(recurse(sample.content, clearFlag(SOURCE_FLAG, opts)), 'sample', opts);
     }
     const dflt = findDefault(e);
     if (dflt) {
-      return wrapResult(dflt.content, 'default', opts);
+      return wrapResult(recurse(dflt.content, clearFlag(SOURCE_FLAG, opts)), 'default', opts);
     }
     if (isFlag(NULLABLE_FLAG, opts)) {
       return wrapResult(null, 'nullable', opts);
     }
+    if (isEnumElement(e)) {
+      const enums = e.enumerations;
+      if (enums && enums.content && enums.content[0]) {
+        return wrapResult(recurse(enums.content[0], clearFlag(SOURCE_FLAG, opts)), 'generated', opts);
```

Comment:
```suggestion
        return wrapResult(recurse(enums.content[0], clearFlag(SOURCE_FLAG, opts)), 'content', opts);
```

Reply:
I am probably wrong on this. Not sure. We might want to forward the value from the enumeration to here, right?

Reply:
`generated` is correct here; `enumerations` describes the type and taking the first of those _is_ generating a value.